RecoJets/JetProducers/plugins/VirtualJetProducer.cc

```diff
@@ -669,87 +668,86 @@ void VirtualJetProducer::writeJets( edm::Event & iEvent, edm::EventSetup const&
   // produce output jet collection
   
   using namespace reco;
-  
-  std::auto_ptr<std::vector<T> > jets(new std::vector<T>() );
-  jets->reserve(fjJets_.size());
+
+  // allocate fjJets_.size() Ts in vector  
+  auto jets = std::make_unique<std::vector<T>>(fjJets_.size());
   
   // Distance between jet centers and overlap area -- for disk-based area calculation
   using RIJ = std::pair<double,double>; 
-  std::vector<std::vector<RIJ> >   rij(fjJets_.size());
+  std::vector<RIJ>   rijStorage(fjJets_.size()*(fjJets_.size()/2));
+  RIJ * rij[fjJets_.size()];
+  unsigned int k=0;
+  for (unsigned int ijet=0;ijet<fjJets_.size();++ijet) {
+     rij[ijet] = &rijStorage[k]; k+=ijet;
+  }
 
   float etaJ[fjJets_.size()],  phiJ[fjJets_.size()];
-  auto etaFromXYZ = [](float x, float y, float z)->float { float t(z/std::sqrt(x*x+y*y)); return vdt::fast_logf(t + std::sqrt(t*t+1.f));};
-  for (auto ijet=0U;ijet<fjJets_.size();++ijet) {
-     float x = fjJets_[ijet].px();
-     float y = fjJets_[ijet].py();
-     float z = fjJets_[ijet].pz();
-     phiJ[ijet] = vdt::fast_atan2(y,x);
-     etaJ[ijet] =etaFromXYZ(x,y,z);
-   } 
+
+  auto orParam_ = 1./rParam_;
```

Comment:
Can protection be added here or in the constructor to ensure that `rParam != 0`? Passing a zero value for this parameter seems like a pretty easy error for someone to make.